algorithms/sorting/quicksort.py

```diff
@@ -0,0 +1,47 @@
+"""Module with the implementation of the QuickSort algorithm."""
+
+
+from typing import List
+from random import randint
+
+from .sort import Sort
+
+
+class QuickSort(Sort):
+    """Class that represents a QuickSort implementation."""
+
+    def _sort(self, items: List) -> List:
+        size = len(items)
+        sorted_items = items.copy()
+
+        self.__sort(sorted_items, 0, size - 1)
+        return sorted_items
+
+    def __sort(self, items: List, p: int, r: int) -> None:
+        if r <= p:
+            return
+
+        q = self.__partition(items, p, r)
+
+        self.__sort(items, p, q)
+        self.__sort(items, q + 1, r)
+
+    def __partition(self, items: List, p: int, r: int) -> int:
```

Comment:
How can we handle different partition strategies here? I notice you implemented the `random pivot` strategy, but there are more strategies that I think we should support when implementing the `QuickSort` algorithm. I will make an emphasis on the partition strategy because I think it will allow comparing how the algorithm behaves differently depending on the chosen strategy.

Reply:
Between the partition strategies of the `QuickSort`, Does only the way of choice the pivot change? if so, I'm thinking about how to abstract the functionality  of choice a partition method using `OOP`, without affecting the other sorting algorithms.

Reply:
I don't know if only the pivot choice is what can vary. But we can start by trying to add the option of modifying just that right now and then keep improving our implementation. I'm also giving it a thought, I will try to come up with a proposal tomorrow. In the meantime, feel free to implement any proposal you think would be fine and we can review it together. Thanks.

Reply:
I implemented the abstraction to get the pivot strategy from the QuickSort's constructor, changed the tests and added a new test for the pivot